### PR TITLE
Handle session expiration gracefully and fix autoLogin redirection 

### LINKED ISSUE #1534: Autologin not working with "jwt expired" error
**What happened**:
The JWT expires very quickly and the user is redirected to the login screen with the corresponding error message even if autoLogin is enabled.

```sh
29 07:34:56.387 http : ::ffff:127.0.0.1 - - [29/Aug/2023:07:34:56 +0000] "GET /auth?redirectUrl=http%3A%2F%2Flocalhost%3A8080%2F HTTP/1.1" 302 724
2023-08-29 07:34:56.392 warn : Authentication failed: UnauthorizedError - jwt expired
2023-08-29 07:34:56.392 http : ::ffff:127.0.0.1 - - [29/Aug/2023:07:34:56 +0000] "POST /auth/token HTTP/1.1" 401 465
2023-08-29 07:34:56.406 http : ::ffff:127.0.0.1 - - [29/Aug/2023:07:34:56 +0000] "GET /auth/logout?error%5Bname%5D=UnauthorizedError&error%5Bmessage%5D=jwt+expired HTTP/1.1" 302 96
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know**:

**Environment**:

### PR DESCRIPTION
**Pull request by holgerkoser**
**What this PR does / why we need it**:
If the login session in the dasboard expires the user no longer gets an error message `JWT expired`. In case of autoLogin is enabled the user is redirected back to the last visited page.

**Which issue(s) this PR fixes**:
Fixes #1534 #865

**Special notes for your reviewer**:

**Release note**:

```bugfix user
If the login session in the dasboard expires the user no longer gets an error message `JWT expired`. In case of autoLogin is enabled the user is redirected back to the last visited page.
```

### CODE REVIEW

File: frontend/src/utils/index.js [hunk 1]
```diff
@@ -287,21 +286,15 @@ export function getTimestampFormatted (timestamp) {
...
 }
 
 export function getTimeStringTo (time, toTime, withoutPrefix = false) {
   if (!time) {
     return undefined
-  } else {
-    if (time.getTime() === toTime.getTime()) {
-      // Equal dates result in text "a few seconds ago", this is not we want here...
-      toTime.setSeconds(toTime.getSeconds() + 1)
-    }
```
Comment (petersutter) [inline frontend/src/utils/index.js:302]: do we have a test for this case?

File: frontend/src/router/guards.js [hunk 2]
```diff
@@ -36,19 +36,42 @@ export function createGlobalBeforeGuards () {
...
-        logger.info('User not found or session has expired --> Redirecting to login page')
-        const query = path !== '/'
-          ? { redirectPath: path }
-          : undefined
-        return {
-          name: 'Login',
-          query,
-        }
+
+      const expiresIn = authnStore.userExpiresAt - Date.now()
```
Comment (petersutter) [inline frontend/src/router/guards.js:54]: shouldn't we rename `userExpiresAt` to `userSessionExpiresAt` or `sessionExpiresAt` to make it clearer?

File: frontend/src/store/authn/helper.js [hunk 3]
```diff
@@ -190,6 +204,8 @@ export function useUserManager (options) {
       let frameRequestCallback
       if (isNoUserError(err)) {
         frameRequestCallback = () => signin()
+      } else if (isSessionExpiredError(err)) {
+        frameRequestCallback = () => signout()
```
Comment (petersutter) [inline frontend/src/store/authn/helper.js:208]: usually this should not happen as we have the expiration timer here, right?https://github.com/gardener/dashboard/blob/2a611f651de356816a85d956a7ef516594c7031b/frontend/src/router/guards.js#L57


In the router guard we pass the `redirectPath` to `signout` when the session expires
https://github.com/gardener/dashboard/blob/2a611f651de356816a85d956a7ef516594c7031b/frontend/src/router/guards.js#L58-L59

but in this case, when the session expires, we don't. 
```js
      } else if (isSessionExpiredError(err)) {
        frameRequestCallback = () => signout()
```

Also, shouldn't we, before we call `signout` or `signin` also log the reason? like we do it [here](https://github.com/gardener/dashboard/blob/2a611f651de356816a85d956a7ef516594c7031b/frontend/src/router/guards.js#L58)

**Review by petersutter — COMMENTED** [review]
very well structured PR 👍

**Comment by petersutter** [comment]
can you also create a hotfix PR? I have already prepared the `hotfix-1.70` branch

File: frontend/src/utils/index.js (hunk 1 above)
Comment (holgerkoser) [reply frontend/src/utils/index.js:302]: I added some tests

File: frontend/src/router/guards.js (hunk 2 above)
Comment (holgerkoser) [reply frontend/src/router/guards.js:54]: renamed to `sessionExpiresAt`

File: frontend/src/store/authn/helper.js (hunk 3 above)
Comment (petersutter) [reply frontend/src/store/authn/helper.js:208]: the error is logged a few lines above.. closing

Comment (holgerkoser) [reply frontend/src/store/authn/helper.js:208]: The error is logged in all cases in the first line of the try catch block. The session expired error should not occur if everthing works as expected. If it happens anyway, it is acceptable to lose the context.

**Review by petersutter — APPROVED** [review]
/lgtm

**Pull request merged**